packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts

```diff
@@ -71,6 +71,10 @@ export function parseCodeBlockTitle(metastring?: string): string {
   return metastring?.match(codeBlockTitleRegex)?.groups!.title ?? '';
 }
 
+export function containsLineNumbers(metastring?: string): boolean {
+  return metastring?.includes('showLineNumbers') || false;
```

Comment:
was thinking maybe we should make it possible to have it enabled by default, and disable on a case-by-case basis?

Reply:
I do not see the need for this, in most cases it is not necessary.

Reply:
Yes, I think opt-in is better than opt-out in most cases. It usually adds unnecessary noise, because unless you want to reference one code line in the main text, you don't need numbering. Another option is to add a default-false theme config option that adds line numbers to all code blocks, but IMO we don't have a strong use-case

Reply:
> Another option is to add a default-false theme config option that adds line numbers to all code blocks, but IMO we don't have a strong use-case

That's what I was thinking about, but we can add later if requested 👍